@xen-orchestra/lite/src/composables/context.composable.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  computed,
+  type ComputedRef,
+  inject,
+  type InjectionKey,
+  type MaybeRefOrGetter,
+  provide,
+  toValue,
+} from "vue";
+
+type Context<T> = {
+  key: InjectionKey<ComputedRef<T>>;
+  defaultValue: T;
+};
+
+export const createContext = <T>(defaultValue: T): Context<T> => {
+  return {
+    key: Symbol() as InjectionKey<ComputedRef<T>>,
+    defaultValue,
+  };
+};
```

Comment:
I think it may be nice to have the possibility to create a more "complex" context, to avoid creating a wrapper as was the case for `ColorContext` and `useColorContext`

Do you think something like this would be possible?
```js
export const createContext = (factory, ...args) => {
  const defaultValue =
    typeof factory === "function" ? factory(...args) : factory;

  return {
    key: Symbol(),
    defaultValue,
  };
};
```
And instead of creating two contexts for the color, do something like:

```js
export const ColorContext = createContext(
  (color: Color) => ({
    value: color,
    bgClass: computed(() => `foobar-${color}`),
    textClass: computed(() => `foobar-text-${color}`),
  }),
  "info"
);

const { value: color, bgClass, textClass } = useContext(ColorContext)